criprobe/cri.py

```diff
@@ -1,14 +1,52 @@
 import re
+import serial
+import serial.tools.list_ports
 
 
 class CriProbe:
 
     def __init__(self, simulated=False):
-        # Perform autodetect here
-        # When no probes are detected we'll create CR-100 and CR-250 simulated probes
-        # these simulated probes will primarily be used for unit testing
+
+        # Autodetects CRI probe/s. When no probe/s detected, creates CR-100 and CR-250 simulated probes.
+
         if simulated:
             self.probes = [{'Type': 'CR-100'},
                            {'Type': 'CR-250'}]
         else:
-            self.probes = None
+            self.probes = []
+            ports = serial.tools.list_ports.comports()
+            for port in ports:
+                if re.search(r'A\d{6}', port.device):
+                    with serial.Serial(port.device, 115200, timeout=1) as cri_probe:
+                        probe_info = {}
+
+                        cri_probe.write(b'RC ID\r\n')
+                        probe_result = cri_probe.readline()
+                        id = re.search(r'(A\d{5})', str(probe_result))
+                        if id:
+                            reg_id = id.group(1)
+                            probe_id = str(reg_id)
+                        probe_info['ID'] = probe_id
```

Comment:
This is not indented enough and would fail if id is not True because probe_id would not yet have a value.